helix-core/src/main/java/org/apache/helix/common/caches/TaskDataCache.java

```diff
@@ -194,9 +194,9 @@ private void refreshContextsAndPreviousAssignments(HelixDataAccessor accessor) {
       }
     }
 
-    List<ZNRecord> contexts = accessor.getBaseDataAccessor().get(contextPaths, null, 0);
+    List<ZNRecord> contexts = accessor.getBaseDataAccessor().get(contextPaths, null, 0, true);
```

Comment:
Does it mean if passing true here, it would throw exception? Then how do we handle this exception?

Reply:
By the way, if passing false here, it would retry this batch async reads infinitely?

Reply:
It will fail the current pipeline, which is what we want. Otherwise, the rebalance will be done based on incomplete information.
If false, it will return partial result.